src/cpp/src/continuous_batching/kvcrush.hpp

```diff
@@ -0,0 +1,59 @@
+// Copyright (C) 2023-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <cmath>
+#include <cstdlib>
+#include <vector>
+
+#include "continuous_batching/attention_output.hpp"
+#include "openvino/genai/cache_eviction.hpp"
+#include "openvino/openvino.hpp"
+
+namespace ov::genai {
+
+class KVCrushAlgorithm {
+public:
+    KVCrushAlgorithm() =
+        default;  // needed only to satisfy DefaultConstructible so that algo objects may be used as values in std::map
+    explicit KVCrushAlgorithm(const KVCrushConfig& eviction_config);
+
+    /** @return A vector of size kvcrush_budget, where each element contains the index of a
+     * representative block selected using KVCrush algorithm */
+    std::vector<std::size_t> get_indices_of_blocks_to_retain_using_kvcrush(
+        size_t decoder_layer_idx,
+        size_t block_size,
```

Comment:
`block_size` is better suited to be a class field initialized via constructor argument since it does not change during algorithm object's lifetime